variants/frontend-react/app/frontend/components/HelloWorld.jsx

```diff
@@ -1,10 +1,26 @@
 import * as PropTypes from 'prop-types';
-import React from 'react';
+import React, { useState } from 'react';
 
-const HelloWorld = ({ greeting }) => <>Greeting: {greeting}</>;
+const HelloWorld = ({ initialGreeting }) => {
+  const [greeting, setGreeting] = useState(initialGreeting);
+
+  function updateGreeting(e) {
```

Comment:
Not the biggest fan of this as a new function will be created every time the component is rendered; would prefer using `useCallback` instead - while overkill in this component, I'd like to encourage its usage for those less familiar with React & hooks.

Also would prefer using arrow functions rather than a hoisted function, to make it easier to reason mentally (since right now its being hoisted to the top of the component, before `setGreeting` is declared).

Finally, would prefer calling the parameter `event` rather than just `e` to be clearer what the expected argument is.

Reply:
`useCallback`: This is an interesting one which I think we've discussed previously. While you're 100% correct from a performance perspective, as you say this level of optimisation is not required for components which are this simple. If we follow that to it's conclusion, we end up with the advice "always memoize your functions" which has been debated with the react community but from everything I've seen, the consensus so far is, "only memoize when the gains are meaningful".

Reply:
That's fine - I'm less concerned about if we use `useCallback` here or not vs using a `const` which I feel makes it a lot more obvious that the functions ref lifecycle is tied to that of the components, by way of the variable being local.

(iirc we discussed this previously too)